.NET/Microsoft.Recognizers.Text.DateTime/Utilities/DurationParsingUtil.cs

```diff
@@ -97,13 +98,57 @@ private static DateObject GetShiftResult(IImmutableDictionary<string, double> ti
             return ret;
         }
 
+        public static DateObject GetNthBusinessDay(DateObject startDate, int n, bool isFuture, out List<DateObject> dateList)
+        {
+            var date = startDate;
+            dateList = new List<DateObject> {date};
+
+            for (var i = 0; i < n; i++)
+            {
+                date = GetNextBusinessDay(date, isFuture);
+                dateList.Add(date);
+            }
+
+            if (!isFuture)
+            {
+                dateList.Reverse();
+            }
+
+            return date;
+        }
+
+        public static DateObject GetNextBusinessDay(DateObject startDate, bool isFuture = true)
+        {
+            var dateIncrement = isFuture ? 1 : -1;
+            var date = startDate.AddDays(dateIncrement);
+            while (date.DayOfWeek == DayOfWeek.Saturday || date.DayOfWeek == DayOfWeek.Sunday)
+            {
+                date = date.AddDays(dateIncrement);
+            }
+
+            return date;
+        }
+
         private static ImmutableDictionary<string, double> ResolveDurationTimex(string timexStr)
         {
             var ret = new Dictionary<string, double>();
-            // resolve duration timex, such as P21DT2H(21 days 2 hours)
+
+            // Resolve duration timex, such as P21DT2H(21 days 2 hours)
             var durationStr = timexStr.Replace("P", "");
             var numberStart = 0;
             var isTime = false;
+
+            // Resolve business days
+            if (durationStr.EndsWith("BD"))
```

Comment:
It's better to define a constant for this term. It will make it much easier to change in the future.